### PR TITLE
Eve-7: Add XZ,YZ, ZX, and ZY projection types. Zeroth version of Lego view type

### PR DESCRIPTION
**Pull request by alja**
## Changes or fixes:
* Add XZ,YZ, ZX, and ZY projection types
* Zero version of Lego view type
* Add energy scaling in REveDataProxyBuilderBase
* Ability to set camera type through REveViewer
* Corrections in reuse of compounds in REveDataSimpleProxyBuilder
* Fix selection/highlight in REveStraightLineSet

## Checklist:

- [ x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

**Comment by alja** [comment]
@linev Please check

**Comment by linev** [comment]
@phsft-bot build

File: graf3d/eve7/src/REveTypes.cxx [hunk 1]
```diff
@@ -45,7 +46,7 @@ REveException REX::operator+(const REveException &s1, ElementId_t x)
 
 REX::RLogChannel &REX::REveLog()
 {
-   static RLogChannel sLog("ROOT.Eve");
+   thread_local RLogChannel sLog("ROOT.Eve");
```
Comment (linev) [inline graf3d/eve7/src/REveTypes.cxx:49]: Are you sure about `thread_local`? 
Idea of log channel that you can configure it from any place - not only from current thread.

**Review by linev — CHANGES_REQUESTED** [review]
I really recommend to use `TString::Format()` and `thread_local` is unclear to me

File: graf3d/eve7/src/REveDataProxyBuilderBase.cxx [hunk 2]
```diff
@@ -205,7 +181,7 @@ REveDataProxyBuilderBase::CreateProduct( const std::string& viewType, const REve
    if (m_collection)
    {
       // debug info in eve browser
-      product->m_elements->SetName(Form("product %s", m_collection->GetCName()));
+      product->m_elements->SetName(Form("product %s viewtype %s", m_collection->GetCName(), viewType.c_str()));
```
[the remark this answers is not in the document]
Comment (alja) [reply graf3d/eve7/src/REveDataProxyBuilderBase.cxx]: I have replaced the Form function with TString::Format.

**Comment by linev** [comment]
@phsft-bot build

File: graf3d/eve7/src/REveDataProxyBuilderBase.cxx [hunk 3]
```diff
@@ -284,7 +260,15 @@ REveDataProxyBuilderBase::SetupElement(REveElement* el, bool color)
    }
 }
 
+//------------------------------------------------------------------------------
 
+void REveDataProxyBuilderBase::ScaleChanged()
+{
+   for (auto &prod : m_products) {
+      if (!HaveSingleProduct())
```
[the remark this answers is not in the document]
Comment (alja) [reply graf3d/eve7/src/REveDataProxyBuilderBase.cxx:268]: The ScaleChanged() function is virtual. 

In CMS event display the scaling have been so far used only for the proxy builder products which have different representation in the different view types. E.g. jet energy scale markers are different in RhoZ or 3D view. I agree this was unnecessary restriction.

**Review by osschar — APPROVED** [review]

File: graf3d/eve7/src/REveTypes.cxx (hunk 1 above)
Comment (alja) [reply graf3d/eve7/src/REveTypes.cxx]: Thank you Sergey! 
I have undone this change.
I have reviewed the logging part again. The thread local is the string that buffer messages for client info:
https://github.com/root-project/root/blob/master/graf3d/eve7/inc/ROOT/REveManager.hxx#L110
 https://github.com/root-project/root/blob/master/graf3d/eve7/src/REveManager.cxx#L1137.

**Comment by linev** [comment]
@alja, please use `TString::Format(,,,).Data()` when need to assign to `std::string`.
On some platforms automatic conversion `TString` -> `std::string` does not work

**Comment by alja** [comment]
@linev I have applied all your corrections.

**Comment by linev** [comment]
@phsft-bot build

**Review by linev — APPROVED** [review]
Now looks ok

**Pull request merged**